_posts/2018-04-25-productivity-with-scripts-screencapture.md

```diff
@@ -0,0 +1,153 @@
+
+I would take a screenshot, find it in my filesystem, open it, edit it, change the name, and then attach it to a message after finding it again in my filesystem. This required a lot of manual steps, and took a decent amount of time. I looked into ways of changing the default name and location of screenshots, but I wanted something a little more custom. After some digging, I found out that macOS has the `screencapture` utility that can be used in `Terminal`. (In your terminal, type `man screencapture` for more information). As soon as I discovered this, I immediately wanted to write a script so I could shave time off this arduous process.
+
+# Investigation Time
+
+After learning of the `screencapture` utility, I began to explore its docs and try it out. I first identified what I wanted to accomplish with my script, and then looked for corresponding flags or variations on the command that would support the behavior I wanted.
+
+I knew I wanted to specify a name and extension for my file on the fly. The following is from `man screencapture`.
+```
+SYNOPSIS
+     screencapture [-SWCTMPcimswxto] file
+```
+I learned `<file>` is the path to destination of where the screenshot will be saved. I also knew that I wanted to select only a portion of the screen with my mouse when taking the screenshot. After exploring the documentation for this utility more I found the following flag.
+```
+-s      Only allow mouse selection mode.
+```
+So I came to the conclusion that I would be invoking the `screencapture` utility from my script as follows.
+```
+screencapture -s <path_to_file>
+```
+I felt comfortable enough with the utility to proceed after identifying how it would be used. All that was left was a matter of writing the bash script and making the file and extension names dynamic.
+
+I also decided that it would be very helpful to have the `<path_to_file>` copied to my clipboard. This would make the file easy to open in preview on the command line after the script ran with the command: `open <path_to_file>`. Having the path to the clipboard would also make it easier to locate in `Finder` for Mac (You can enter absolute path in `Finder` with `command` + `shift` + `G`).
+Luckily, Mac has the `pbcopy` command that allows you to paste to your system `Clipboard` via the command line. You can read more about `pbcopy` by typing `man pbcopy` in your terminal.
+
+# Writing The Script
+
+I will not go a ton into bash scripting in this tutorial. There are plenty of other great tutorials and resources for that online. However, I will provide a brief description of my thought process and detailed comments within the attached snippet for the script.
```

Comment:
~I will not go a ton into bash scripting in this tutorial. There are plenty of other great tutorials and resources for that online. However, I will provide a brief description of my thought process and detailed comments within the attached snippet for the script.~

Just do it!